exporters/prometheus/test/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+foreach (
+        testname
+        prometheus_exporter_utils_test
+        prometheus_collector_test
+        prometheus_exporter_test)
+    add_executable(${testname} "${testname}.cc")
+    target_link_libraries(
+            ${testname} ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
+            prometheus_exporter prometheus-cpp::pull)
+    gtest_add_tests(TARGET ${testname} TEST_PREFIX exporter. TEST_LIST ${testname})
```

Comment:
Could you combine this unit tests with the piece of code that is tested into one PR? And provide Bazel build support for it, as well as hooking it into the global CMake build by adding the prometheus directory as a subdirectory to `exporters/CMakeLists.txt`?

I would greatly help the reviewing process to have those tests running in the CI.

Reply:
Yes, I can do that, but I am worrying that the combined PR will be so big. We have three unit test files that contain more than 1000 lines. Can we have such a large PR?

Reply:
I think it's good to have each PR buildable, with unit tests that cover the added new functionality. If the only way to ensure that is a bigger PR, than I think it's better to have a bigger PR.

Maybe you can have three PRs: first the exporter utils with unit tests, second the collector with unit tests based on the first, and third the exporter with unit tests based on the second. We can then review and merge from first to third, rebasing later ones when previous ones got merged.